tools/IndexStore/CMakeLists.txt

```diff
@@ -34,32 +34,24 @@ add_clang_library(IndexStore ${ENABLE_SHARED} ${ENABLE_STATIC}
   Core
   Support
   )
+set_target_properties(IndexStore
+                      PROPERTIES
+                        CXX_VISIBILITY_PRESET hidden
+                        CXX_VISIBILITY_INLINES_HIDDEN YES)
```

Comment:
I think I wasn't clear in my previous comments. Setting these compiler arguments only for the files of `IndexStore` doesn't work because all the symbols coming from rest of llvm/clang/swift are still having 'default' visibility. These arguments are useful only if you can apply them across everything, so visibility is affected for every symbol.

If you want to see for yourself that this is not working for reducing the exported symbols, try this. Build `libIndexStore.dylib` on Darwin without using an exports list and then run this command on the dylib:

`nm -g /path/to/lib/libIndexStore.dylib | grep ' T '`

this command will show you what symbols get exported by the dylib. Without the export list you'll see a huge number of C++ symbols coming from llvm/clang/swift.